packages/mds-ingest-service/service/validators.ts

```diff
@@ -126,24 +108,56 @@ export const { validate: validateTelemetryDomainModel, $schema: TelemetrySchema
     { useDefaults: true }
   )
 
-export const { validate: validateEventDomainModel, isValid: isValidEventDomainModel } =
-  schemaValidator<DeviceDomainModel>(
-    Joi.object<EventDomainModel>()
-      .keys({
-        device_id: Joi.string().uuid().required(),
-        provider_id: Joi.string().uuid().required(),
-        timestamp: Joi.number().required(),
-        event_types: Joi.array()
-          .valid(Joi.string().valid(...VEHICLE_EVENTS))
-          .required(),
-        vehicle_state: Joi.string().valid(...VEHICLE_STATES),
-        telemetry_timestamp: Joi.number().allow(null),
-        telemetry: telemetrySchema.allow(null),
-        trip_id: Joi.string().uuid().allow(null),
-        service_area_id: Joi.string().uuid().allow(null)
-      })
-      .unknown(false)
-  )
+export const { validate: validateEventDomainModel, $schema: EventSchema } = SchemaValidator<EventDomainModel>(
+  {
+    $id: 'Event',
+    type: 'object',
+    properties: {
+      device_id: uuidSchema,
+      provider_id: uuidSchema,
+      timestamp: { type: 'integer' },
+      event_types: {
+        type: 'array',
+        items: {
+          type: 'string',
+          enum: [...new Set(VEHICLE_EVENTS)]
+        },
+        minItems: 1
+      },
+      vehicle_state: {
+        type: 'string',
+        enum: [...new Set(VEHICLE_STATES)]
+      },
+      telemetry: TelemetrySchema /** NOTE:
+        Telemetry is considered non-optional as of MDS 1.0, even though some legacy events do not have telemetry (e.g. `register`).
+        This is why the schema is more restrictive (non-nullable) than the typdef.
+       */,
```

Comment:
Nit but can you move the comma to be in between `TelemetrySchema` and your comment?

Reply:
Yeah, I might just put the comment above the schema def. I wanna make sure it's clear that the comment goes along with that bit

Reply:
Fixed in da771e6